addStudy.py

```diff
@@ -0,0 +1,101 @@
+# addStudy.py
+
+#!/usr/bin/python3
+
+# python addStudy.py -c rwa_configuration.yml -l
+# python addStudy.py -c rwa_configuration.yml -a "studyName"
+
+# addStudy.py
+#
+# Adding new studies to the database
+
+# Import our libraries
+import os
+import sys
+import argparse
+import yaml
+import psycopg2
+from include.calibrationLib import *
+
+# List all studies
+def get_studies(configuration, studName):
+    # return "List to populate"
+    sql = '''SELECT * FROM STUDY'''
+
+    return select(configuration["connection_string"], sql, {'name': None})
+
+# add studies
+def add_file(configuration, Name):
+    Name = str(Name)
+
+    sql = '''INSERT INTO study(Name) VALUES(%s) RETURNING id;'''
+
+    return select(configuration["connection_string"], sql, {'Name': Name})
+
+# connection
+def select(connectionString, sql, parameters):
+    # Open the connection
+    connection = psycopg2.connect(connectionString)
+
+    cursor = connection.cursor()
+    # add study
+    if (sql.find("INSERT") == 0):
+        cursor.execute(sql, (parameters['Name'],))
+        study_id = cursor.fetchone()[0]
+
+        connection.commit()
+
+        cursor.close()
+
+        return study_id
+    # list all studies
+    if (sql.find("SELECT") == 0):
+        cursor.execute(sql)
+
+        rows = cursor.fetchall()
+        for row in rows:
+            print(row)
+            # print(sql.find("SELECT"))
+        cursor.close()
+
+
+def main(configuration, flag, studyId):
+    # Loading the configuration
+    cfg = load_configuration(configuration)
+    if (flag == 0):
+        get_studies(cfg, studyId)
+    if (flag == 1):
+        studies = add_file(cfg, studyId)
+        print("Study Id: " + str(studies))
```

Comment:
This way of parsing out what operation to do works in simple cases like this, but over the long run it would require a lot of refactoring. Best to avoid that now by making sure that the `main` function is appropriately flexible. Remember, an option is to just shift everything under `__name__` into the main function!